### ISSUE TITLE
Sorry, this binary distribution does not support installing sage packages

### ISSUE THREAD

**Issue by Ubiquitous** [3-manifolds/Sage_macOS#72]
Attempting to install `database_cremona_ellcurve`

```
 % sage -i database_cremona_ellcurve
Sorry, this binary distribution does not support installing sage packages.
If database_cremona_ellcurve is available from pypi you may install it by running
    %pip install database_cremona_ellcurve
from the sage prompt.
```

Trying with pip

```
ERROR: Could not find a version that satisfies the requirement database_cremona_ellcurve (from versions: none)
ERROR: No matching distribution found for database_cremona_ellcurve
```

What is the recommended solution? 

Using [Sage version 10.4](https://github.com/3-manifolds/Sage_macOS/releases/tag/v2.4.0) arm64

**Comment by culler** [comment]
Hmmm.  I think there is no recommended solution at this moment.  We chose not to include big databases in the app bundle in order to keep the download size reasonable.  (1GB is already a stretch.).

But apparently there is no pip wheel for the Cremona elliptic curve databases.  I don't know why not.  Naively, that would seem like an ideal way to distribute them.  We have pip packages for SnapPy's databases, which are also sqlite files.  I will ask around and try to find out what the obstacles are to doing that.

**Comment by Ubiquitous** [comment]
I can understand not including it in the app bundle, although not being able to install sage packages due to the binary distribution not supporting it completely limits the capability. Do you know why that is? I appreciate you asking around and hopefully you will have a resolution soon ~ thank you.

**Comment by neldredge-unco** [comment]
Along similar lines, is it possible to install fricas?  There likewise doesn't seem to be a pip package for that.

**Comment by culler** [comment]
I can add fricas.  I will try adding the Cremona database as well and see how much it increases the download size by.  I don't think that a pip package will exist in time for the 10.5 release.

**Comment by culler** [comment]
> not being able to install sage packages due to the binary distribution not supporting it completely limits the capability. Do you know why that is? 

The SageMath app is signed and notarized.  If you install an additional Sage package after notarization it breaks the signature and, since the app is notarized, Apple will not allow it to run.  Notarization is essential if you want to have an installation process that works smoothly for normal users.

**Comment by culler** [comment]
Including Fricas adds 12 minutes to the build time.  Adding both Fricas and the Cremona Database increases the download size from 1 GB to 1.3 GB.  I will need to think about whether that is worth it.

**Comment by culler** [comment]
I have decided two things:
1. Adding Fricas adds no value to the app.  You can download Fricas from [sourceforge](https://sourceforge.net/projects/fricas/files/) and install it on your system.  As long as it is in your path, the Sage interface to Fricas will work.  As far as I know, installing the Sage package does not add any functionality beyond that.  And the extra build time and larger download size outweigh the convenience for the few people who want to use it.  (I.e 1 request after more than 20,000 downloads.
2. I will pursue a workaround for the Cremona database other than creating a Pypi package.  I think the sqlite file could be installed outside the app bundle and the Sage package could be patched to allow use of an external database file if one exists.

**Comment by neldredge-unco** [comment]
Re Fricas: I can accept that.  Thanks anyway for considering it.  Mostly I just wish there was a Homebrew package for Fricas, and then it really would be easy.

**Comment by culler** [comment]
@Ubiquitous: I am finally able to provide a recommended solution.

If you would like to use the large Cremona database with the soon-to-be-released SageMath-10.5, all you need to do is to download the database file and move it to ~/.sage/db/cremona.

```
cd /tmp
curl -L -O https://github.com/sagemath/sage/releases/download/10.5/database_cremona_ellcurve-20190911.tar.bz2
tar xfj database_cremona_ellcurve-20190911.tar.bz2
mkdir -p ~/.sage/db/cremona
mv src/cremona.db ~/.sage/db/cremona
rm database_cremona_ellcurve-20190911.tar.bz2 

``` 

Sage will then detect that the database file exists and use it in place of the mini cremona database which is included in the application bundle.  You can check if it worked like this:
```
sage: DB = CremonaDatabase()
sage: type(DB)
<class 'sage.databases.cremona.LargeCremonaDatabase'>
```

**Comment by neldredge-unco** [comment]
In case anyone comes across my earlier comment about fricas: There is now a [homebrew formula for fricas](https://formulae.brew.sh/formula/fricas).  After installing it with `brew install fricas`, I had to add `export PATH="$PATH:/opt/homebrew/bin"` to `~/.sage/sagerc`, shut down all kernels, and restart the SageMath application.  Now fricas works inside Sage.
